src/main/java/org/jlab/myquery/MySamplerController.java

```diff
@@ -167,83 +152,154 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response)
-                    } else if (stream.getType() == MultiStringEvent.class) {
-                        dataLength = generateMultiStringStream(gen, (EventStream<MultiStringEvent>) stream, formatAsMillisSinceEpoch, adjustMillisWithServerOffset,
-                                timestampFormatter);
-                    } else {
-                        throw new ServletException("Unsupported data type: " + stream.getClass());
-                    }
-                    gen.writeEnd();
 
-                    gen.write("returnCount", dataLength);
-                }
+            }
+        } catch (Exception ex) {
+            response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+            LOGGER.log(Level.SEVERE, "Unexepected error", ex);
+            throw ex;
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private boolean processChannelRequest(MySamplerWebService service, String deployment, JsonGenerator gen, String channel,
+                                          Instant begin, long intervalMillis, long sampleCount, boolean updatesOnly,
+                                          boolean formatAsMillisSinceEpoch, boolean adjustMillisWithServerOffset,
+                                          DateTimeFormatter timestampFormatter, DecimalFormat decimalFormatter,
+                                          boolean enumsAsStrings) throws ServletException {
+        gen.writeStartObject(channel);
+        boolean error = false;
+
+        // Write out the channel's metadata.
+        Metadata metadata = null;
+        try {
+            metadata = service.findMetadata(channel);
+            if (metadata == null) {
+                throw new Exception("Unable to find channel: '" + channel + "' in deployment: '" + deployment + "'");
+            }
+            writeMetadata("metadata", gen, metadata);
+        } catch (Exception ex) {
+            gen.write("error", ex.getMessage());
+            gen.writeEnd();
+            return true;
+        }
+
+        // Write out the channels enum label's if channel was an enumerated type
+        List<ExtraInfo> enumLabels = null;
+        if (metadata.getMyaType() == MyaDataType.DBR_ENUM) {
+            try {
+                enumLabels = service.findExtraInfo(metadata, "enum_strings", begin,
+                        begin.plusMillis(intervalMillis * (sampleCount - 1)));
+                writeEnumLabels("labels", gen, enumLabels, formatAsMillisSinceEpoch, adjustMillisWithServerOffset,
+                        timestampFormatter);
+            } catch (Exception ex) {
+                gen.write("error", ex.getMessage());
                 gen.writeEnd();
+                return true;
+            }
+        }
 
-                gen.flush();
+        // Write out the channel's data
+        EventStream stream = null;
+        try {
+
+            // Cannot use try with resources since we may sometimes wrap stream in a LabeledEnumStream
```

Comment:
This caught my eye.  Not immediately obvious why can't rely on [WrappedStream](https://github.com/JeffersonLab/jmyapi/blob/ae2b18a59d66ca33b4493c218746366366e00508/src/main/java/org/jlab/mya/stream/WrappedStream.java#L47) to handle close

Reply:
The problem is that variable assignment done in try-with-resources must be (effectively) final.  Since we want to sometimes update the stream variable, it can't be final.